### ISSUE TITLE
Permission error: permission errono13

### ISSUE THREAD

**Issue by panrenzaishu** [Ecogenomics/CheckM#282]
Hi, I encountered a problem in using CheckM and checked is embeded in the Atlas pileflow made by snakemake. The program is not run on personl computer but on orgization high performance computer. Here, i post the error issue: 
Unexpected error: <class 'PermissionError'>
Traceback (most recent call last):
  File "/scratch/panchaozhi/atlas/databases/conda_envs/d94ab96d/bin/checkm", line 611, in <module>
    checkmParser.parseOptions(args)
  File "/scratch/panchaozhi/atlas/databases/conda_envs/d94ab96d/lib/python3.8/site-packages/checkm/main.py", line 858, in parseOptions
    self.tree(options)
  File "/scratch/panchaozhi/atlas/databases/conda_envs/d94ab96d/lib/python3.8/site-packages/checkm/main.py", line 119, in tree
    binIdToModels = mgf.find(binFiles,
  File "/scratch/panchaozhi/atlas/databases/conda_envs/d94ab96d/lib/python3.8/site-packages/checkm/markerGeneFinder.py", line 58, in find
    workerQueue = mp.Queue()
  File "/scratch/panchaozhi/atlas/databases/conda_envs/d94ab96d/lib/python3.8/multiprocessing/context.py", line 103, in Queue
    return Queue(maxsize, ctx=self.get_context())
  File "/scratch/panchaozhi/atlas/databases/conda_envs/d94ab96d/lib/python3.8/multiprocessing/queues.py", line 42, in __init__
    self._rlock = ctx.Lock()
  File "/scratch/panchaozhi/atlas/databases/conda_envs/d94ab96d/lib/python3.8/multiprocessing/context.py", line 68, in Lock
    return Lock(ctx=self.get_context())
  File "/scratch/panchaozhi/atlas/databases/conda_envs/d94ab96d/lib/python3.8/multiprocessing/synchronize.py", line 162, in __init__
    SemLock.__init__(self, SEMAPHORE, 1, 1, ctx=ctx)
  File "/scratch/panchaozhi/atlas/databases/conda_envs/d94ab96d/lib/python3.8/multiprocessing/synchronize.py", line 57, in __init__
    sl = self._semlock = _multiprocessing.SemLock(
PermissionError: [Errno 13] Permission denied

It should not be the dataset issue, because i found below from initialize_checkm.log and the data setRoot has the right directory:
[2020-12-25 19:02:00] INFO: CheckM v1.1.2
[2020-12-25 19:02:00] INFO: checkm data setRoot /scratch/panchaozhi/atlas/databases/checkm
[2020-12-25 19:02:00] INFO: [CheckM - data] Check for database updates. [setRoot]

Path [/scratch/panchaozhi/atlas/databases/checkm] exists and you have permission to write to this folder.
(re) creating manifest file (please be patient).

Path [/scratch/panchaozhi/atlas/databases/checkm] exists and you have permission to write to this folder.
(re) creating manifest file (please be patient).

I am looking for to the reply. Thank you very much!

**Comment by dparks1134** [comment]
Hi. Thank you for your email. I'm not sure what to suggest as a potential solution. I'm not familiar with he `Atlas pileflow made by snakemake` as this isn't part of CheckM packaging I provide.

**Comment by panrenzaishu** [comment]
Hi, Thank you for your rapid response. I try to re-install the tool again
to see whether the issue is resolved. Possibly due to recent computer
updating and annual maintenance. Thank you very much. Happy new year!

**Comment by SilasK** [comment]
Atlas just runs checkM installed with conda.

@dparks1134 Do you have an idea what the error means.
```
sl = self._semlock = _multiprocessing.SemLock(
PermissionError: [Errno 13] Permission denied
```

Could it be related to python 3.8?

**Comment by SilasK** [comment]
Do you know if checkM acces the shared memory `/dev/shm`or something like that?

**Comment by SilasK** [comment]
It seems that checkm doesn't work always ideal with python 3.8

**Comment by SilasK** [comment]
We solved the issue by using python 3.6. with checkM
